### ISSUE TITLE
Bug - Verification phone format

### ISSUE THREAD

**Issue by npracht** [mautic/mautic#1614]
## Description
The phone format field is not controlling correctly the values written in it.
I can for instance write values with letters.

## Step to reproduce
1. Create a form
2. Add a field with phone format
3. Submit the form with any value

**Comment by escopecz** [comment]
The phone format is quite different in different countries. I've seen a phone number with letters as well (crazy Americans). I think that all that the Phone field does is that the input is type of tel. It will tell the browser that the user will input a phone number in it. For example the mobile browsers should open the number keyboard instead of the letter keyboard for this input.

So how exactly would you want to implement the phone number validation?

**Comment by npracht** [comment]
I dunno, i think Synfony may have a phone number control format.
I mean when you create an account on google they have the control and it should correspond to any international number format.

N.

**Comment by npracht** [comment]
Moreover it would be great to have this control for the SMS feature that's gonna come, otherwise it's gonna be a mess :-)

Refers to that PR - https://github.com/mautic/mautic/pull/1525

**Comment by martinstr** [comment]
Twilio have a section about what format they use https://support.twilio.com/hc/en-us/articles/223183008-Formatting-International-Phone-Numbers
Don't know but something like this should be implemented into the phone field https://github.com/jackocnr/intl-tel-input

**Comment by edudemy** [comment]
To validate phone numbers, we could use Google's libphonenumber library.
https://github.com/googlei18n/libphonenumber
https://github.com/giggsey/libphonenumber-for-php

**Comment by npracht** [comment]
Yes google lib seems the most robust and that been integrated in a Symfony 2 bundle: https://github.com/misd-service-development/phone-number-bundle

This is critical because withtout that structure SMS cannot go to Twilio + it is a basic standard in the software industry.

**Comment by kuzmany** [comment]
@escopecz I am going to work on this request in next days.
Do you have some suggestions (or somebody else from Mautic) before I'll start on it? 
I did research: 
1. https://github.com/misd-service-development/phone-number-bundle  - littleheavy, but looks as simple  bundle to Mautic - maybe app/bundles/PhoneNumberBundle? What do you think?
2. https://www.twilio.com/blog/2016/03/how-to-validate-phone-numbers-in-php-with-the-twilio-lookup-api.html validation via Twillio Lookup service (but it's paid $0.005/per lookup)

What should be for Mautic easier?
I guess Mautic validate phone number during the send text message (maybe i never used that).  
This is validation for forms.

**Comment by escopecz** [comment]
If there is a PHP library which can take care of it, use it. But it should be added as a Composer dependency and live in the vendor directory. Then use it in the LeadBundle and FormBundle to validate the fields type of phone.

I think Mautic users would prefer that over a paid service. Maybe something like https://packagist.org/packages/giggsey/libphonenumber-for-php ?

**Comment by kuzmany** [comment]
@escopecz thanks. Work in progress 👍

**Comment by kuzmany** [comment]
Let's test guys https://github.com/mautic/mautic/pull/5400

**Comment by zaharovrd** [comment]
how validate phone number as +7 (999) 999-99-99  or  +9 (999) 999-999-999
that is, just filling with the same numbers

**Comment by npracht** [comment]
Hi @zaharovrd, not sure to understand your question.
We use this library to validate phone number on international format (you can enable it on the contact field setting (phone) or on the form field).
https://github.com/giggsey/libphonenumber-for-php

This is based on https://github.com/google/libphonenumber

**Comment by zaharovrd** [comment]
> Hi @zaharovrd, not sure to understand your question. We use this library to validate phone number on international format (you can enable it on the contact field setting (phone) or on the form field). https://github.com/giggsey/libphonenumber-for-php
> 
> This is based on https://github.com/google/libphonenumber

Hello!
Yes, I am aware of this library, but I did not find in it a check for phone numbers filled with the same numbers, such as
99999999999 or 1111111111
It’s unlikely that there may be such a number, but they are often used to complete the form, not wanting to leave the real number

**Comment by npracht** [comment]
I don't think this is possible to handle. That will work on Google forms as well. If Google cannot avoid it.... i don't think we can :D
